src/main/java/org/openrewrite/java/template/processor/RefasterTemplateProcessor.java

```diff
@@ -306,12 +309,15 @@ public void visitClassDef(JCTree.JCClassDecl classDecl) {
                 if (classDecl.sym != null && classDecl.sym.getNestingKind() == NestingKind.TOP_LEVEL && !recipes.isEmpty()) {
                     boolean outerClassRequired = descriptor == null;
                     try {
+                        Symbol.PackageSymbol pkg = classDecl.sym.packge();
```

Comment:
Deduplicating `classDecl.sym.packge()` and `classDecl.sym.fullname.toString()` at the top of this method would yield an NPE, unless more things are refactored. But that would likely change the indentation of most of the code in this method, which would make the PR way harder to review. Seems better to refactor that separately (if at all).